Examples/Algorithms/Fatras/src/FatrasAlgorithm.cpp

```diff
@@ -166,19 +167,8 @@ ActsExamples::FatrasAlgorithm::FatrasAlgorithm(Config cfg,
   ACTS_DEBUG("hits on passive surfaces: " << m_cfg.generateHitsOnPassive);
 
   // construct the simulation for the specific magnetic field
-  m_sim = std::visit(
-      [&](auto &&mf)
-          -> std::unique_ptr<ActsExamples::detail::FatrasAlgorithmSimulation> {
-        // the variant sub-types are always wrapped in a shared_ptr
-        using ThisMagneticField =
-            typename std::decay_t<decltype(mf)>::element_type;
-        using SharedMagneticField = Acts::SharedBField<ThisMagneticField>;
-        using Implementation = FatrasAlgorithmSimulationT<SharedMagneticField>;
-
-        return std::make_unique<Implementation>(
-            SharedMagneticField(std::move(mf)), m_cfg, lvl);
-      },
-      m_cfg.magneticField);
+  m_sim = std::make_unique<FatrasAlgorithmSimulationT>(m_cfg.magneticField,
```

Comment:
Ok, that's really nicer now.

Reply:
I think this whole level of wrapping can go, now that `FatrasAlgorithmSimulationT` isn't even a template anymore. I'll try to do a pass over the Fatras example in the future to throw out the now unnecessary complexity.